Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -448,21 +461,38 @@ void FunctionDefinitionWriter::visit(AST::CallExpression& call)
                 m_stringBuilder.append(m_indent);
                 visit(argument);
                 m_stringBuilder.append(",\n");
-                first = false;
             }
         }
         m_stringBuilder.append(m_indent, "}");
-    } else {
-        visit(call.target());
-        m_stringBuilder.append("(");
-        for (auto& argument : call.arguments()) {
-            if (!first)
-                m_stringBuilder.append(", ");
-            visit(argument);
-            first = false;
+        return;
+    }
+
+    if (is<AST::NamedTypeName>(call.target())) {
+#define ALIAS(alias) \
+    [](FunctionDefinitionWriter* writer, AST::CallExpression& call) { \
+        writer->m_stringBuilder.append(alias); \
+        visitArguments(writer, call, 1); \
+    }
+
+        static constexpr std::pair<ComparableASCIILiteral, void(*)(FunctionDefinitionWriter*, AST::CallExpression&)> builtinMappings[] {
+            { "textureSample", [](FunctionDefinitionWriter* writer, AST::CallExpression& call) {
+                ASSERT(call.arguments().size() > 1);
+                writer->visit(call.arguments()[0]);
+                writer->m_stringBuilder.append(".sample");
+                visitArguments(writer, call, 1);
+            } },
+            { "vec4", ALIAS("float4") },
```

Comment:
The macro is only used twice :/

What else will we use it for in the future?

I don't think this needs to be a macro. Can we just pass the string into a lambda?

Reply:
The macro is only used once now, but it would be used in the future for all the types that have a different name in Metal vs WGSL (at least all the vector and matrix types I believe). I don't think we can pass the string to a lambda, or at least I couldn't find a way to do it, since the lambda has to be constexpr and can't capture anything.